cli/command/service/create.go

```diff
@@ -106,6 +106,11 @@ func runCreate(dockerCli *command.DockerCli, flags *pflag.FlagSet, opts *service
 		createOpts.EncodedRegistryAuth = encodedAuth
 	}
 
+	// query registry if flag disabling it was not set
+	if !opts.noResolveImage {
```

Comment:
hm, interesting; have you tried debugging (logging/printing the value of `opts.noResolveImage`) just for fun?